src/internal/server/PrintLeftoverLabelHandler_test.go

```diff
@@ -0,0 +1,134 @@
+package server_test
+
+import (
+	"bytes"
+	"io"
+	"main/internal/server"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+type testParameters_PrintLeftoverLabelHandler struct {
+	reqMethod          string
+	reqBody            io.Reader
+	expectedStatusCode int
+	expectedMessage    string
+}
+
+func TestPrintLeftoverLabelHandler(t *testing.T) {
+	// define test cases
+	var testParams = []testParameters_PrintLeftoverLabelHandler{
+		// should fail because incorrect HTTP method
+		{
+			reqMethod:          "GET",
+			reqBody:            nil,
+			expectedStatusCode: http.StatusBadRequest,
+			expectedMessage:    "This endpoint only supports POST requests\n",
+		},
+		// should fail because incorrect HTTP method
+		{
+			reqMethod:          "PUT",
+			reqBody:            nil,
+			expectedStatusCode: http.StatusBadRequest,
+			expectedMessage:    "This endpoint only supports POST requests\n",
+		},
+		// should fail because incorrect HTTP method
+		{
+			reqMethod:          "CHANGE",
+			reqBody:            nil,
+			expectedStatusCode: http.StatusBadRequest,
+			expectedMessage:    "This endpoint only supports POST requests\n",
+		},
+		// should fail because the body is malformed
+		{
+			reqMethod:          "POST",
+			reqBody:            bytes.NewBufferString("some text"),
+			expectedStatusCode: http.StatusBadRequest,
+			expectedMessage:    "Malformed request body\n",
+		},
+		// should fail because the body is missing a required field
+		{
+			reqMethod:          "POST",
+			reqBody:            bytes.NewBufferString(`{"labelText":"test label text"}`),
+			expectedStatusCode: http.StatusBadRequest,
+			expectedMessage:    "Malformed request body\n",
+		},
+		// should fail because the body is missing a required field
+		{
+			reqMethod:          "POST",
+			reqBody:            bytes.NewBufferString(`{"quantity":2}`),
+			expectedStatusCode: http.StatusBadRequest,
+			expectedMessage:    "Malformed request body\n",
+		},
+		// should fail because a payload field has an incorrect type
+		{
+			reqMethod:          "POST",
+			reqBody:            bytes.NewBufferString(`{"labelText":2,"quantity":2}`),
+			expectedStatusCode: http.StatusBadRequest,
+			expectedMessage:    "Malformed request body\n",
+		},
+		// should fail because payload is too large
+		{
+			reqMethod:          "POST",
+			reqBody:            bytes.NewBufferString(`{"labelText":"Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa.","quantity":2}`),
+			expectedStatusCode: http.StatusRequestEntityTooLarge,
+			expectedMessage:    "Request body is too large\n",
+		},
+		// should pass
+		{
+			reqMethod:          "POST",
+			reqBody:            bytes.NewBufferString(`{"labelText":"test label text","quantity":2}`),
+			expectedStatusCode: http.StatusOK,
+			expectedMessage:    `{"status":"success"}`,
+		},
```

Comment:
You could also add a test to see what happens if pdf generation fails. It forces you to mock pdf generation

Reply:
#5